src/regdesk/workflow_button.js

```diff
@@ -0,0 +1,100 @@
+// eslint-disable-next-line no-unused-vars
+import { RegState } from './states/reg_state.js';
+
+/**
+ * Wrapper for one of the side gutter buttons for navigating workflows
+ */
+export class WorkflowButton {
+  #element;
+
+  /**
+   * @type {RegState}
+   */
+  #transitionEventTarget;
+  #transitionEvent;
+  #transitionEventDetails;
+
+  #buttonClickEvent = 'click';
+
+  /**
+   * Initializes a new instance of the WorkflowButton class.
+   * @param {Element} buttonElement - The button to wrap
+   */
+  constructor(buttonElement) {
+    this.#element = buttonElement;
+  }
+
+  /**
+   * Hide this button.
+   * @return {WorkflowButton} - This button.
+   */
+  hide() {
+    this.#element.classList.remove('visible');
+    this.#element.classList.add('invisible');
+    return this;
+  }
+
+  /**
+   * Show this button
+   * @return {WorkflowButton} - This button.
+   */
+  show() {
+    this.#element.classList.remove('invisible');
+    this.#element.classList.add('visible');
+    return this;
+  }
+
+  /**
+   * Remove the transition callback registered on the button.
+   * @return {WorkflowButton} - This button.
+   */
+  clearTransitionCallback() {
+    this.#transitionEventTarget = undefined;
+    this.#transitionEvent = undefined;
+    this.#transitionEventDetails = undefined;
+    this.#element.removeEventListener(this.#buttonClickEvent, this.handleClickEvent);
+    return this;
+  }
+
+  /**
+   * Register a transition event for this button to fire on a click.
+   * Overwrites any existing transitions.
+   * @param {RegState} transitionEventSender - The object to send the event through
+   * @param {string} event - The event name to send
+   * @param {*} details - Optional details object to add to the event.
+   * @return {WorkflowButton} - This button.
+   */
+  setTransitionCallback(transitionEventSender, event, details) {
+    this.clearTransitionCallback();
+    this.#transitionEventTarget = transitionEventSender;
+    this.#transitionEvent = event;
+    this.#transitionEventDetails = details;
+    this.#element.addEventListener(this.#buttonClickEvent, this.handleClickEvent.bind(this));
+    return this;
+  }
+
+  /**
+   * Dispatch a click event, if relevant to do so.
+   *
+   * @description Okay but why though? So we can remove this event listener.
```

Comment:
Haha, agreed, feels a little funny.

Reply:
Man I spent so much time trying to find a better system for re-using UI elements with different callbacks.